### PR TITLE
Introduce Instrument endpoint

### PR DESCRIPTION
**Pull request by konstantinos-kafkas-cko**
- Introducing the new instrument endpoint specification

A valid token regex for UPAPI is → `"^(tok)_(\w{26})$"`
A valid token regex for Classic API is → `"^(card_tok)_(\w{12})$"`

### CODE REVIEW

File: spec/swagger.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ tags:
     description: Create a payment source for a customer that you can use for one or more payments.
   - name: Tokens
     description: Create a token that represents a card's details (or their tokenized form in a digital wallet) that you can later use to request a payment, without you having to process or store any sensitive information.
+  - name: Instruments
+    description: Create an instrument source that you can use for one or more payments.
```
Comment (ramani-brown-cko) [inline spec/swagger.yaml]: Can we change this to: 
Create a payment instrument that can be used as the source or destination for future payments

File: spec/paths/instruments.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+post:
+  tags:
+    - Instruments
+  summary: Create Instrument
+  description: |
+    Create instrument using a Checkout.com token
```
Comment (ramani-brown-cko) [inline spec/paths/instruments.yaml]: Can we change this to:
Create a payment instrument from a Checkout.com token.

File: spec/components/schemas/Instruments/InstrumentResponse.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        example: "VISA"
+    last4:
+        type: string
+        description: The last four digits of the card number
+        example: "9996"
+        minLength: 4
+        maxLength: 4
+    bin:
+        type: string
+        description: The card issuer's BIN
```
Comment (ramani-brown-cko) [inline spec/components/schemas/Instruments/InstrumentResponse.yaml]: Can we change this to:
The card issuer's bank identification number (BIN)

File: spec/components/schemas/Instruments/InstrumentResponse.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        example: Consumer
+    issuer:
+        type: string
+        description: The name of the card issuer
+        example: "GOTHAM STATE BANK"
+    issuer_country:
+        type: string
+        maxLength: 2
+        minLength: 2
+        description: The card issuer country ISO-2 code
```
Comment (ramani-brown-cko) [inline spec/components/schemas/Instruments/InstrumentResponse.yaml]: hyperlink "ISO-2 code" to https://docs.checkout.com/docs/country-codes

File: spec/components/schemas/Instruments/InstrumentResponse.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        maxLength: 4
+    bin:
+        type: string
+        description: The card issuer's BIN
+        example: "454347"
+        maxLength: 4
+    card_type:
+        type: string
+        description: The card type
+        enum:
```
Comment (ramani-brown-cko) [inline spec/components/schemas/Instruments/InstrumentResponse.yaml]: Potentially need to add "Charge" to the enum values - I will confirm tomorrow

File: spec/components/schemas/Instruments/InstrumentRequest.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  - token
+properties:
+  type:
+    type: string
+    description: The instrument type
+    pattern: token
+    example: token
+  token:
+    type: string
+    description: The Checkout.com token
```
[the remark this answers is not in the document]
Comment (konstantinos-kafkas-cko) [reply spec/components/schemas/Instruments/InstrumentRequest.yaml]: @mathew-tuvesson-cko asked me to not have `(e.g., a card or digital wallet token)` in the description

[the remark this answers is not in the document]
Comment (ramani-brown-cko) [reply spec/components/schemas/Instruments/InstrumentRequest.yaml]: No probs - i just copied it from the UPAPI one

**Comment by konstantinos-mastropavlos-cko** [comment]
Please do not merge this before #99

**Pull request merged**